modules/tectonic/resources/tectonic.sh

```diff
@@ -1,30 +1,20 @@
-#!/bin/sh
+#!/usr/bin/env bash
 set -e
 
 KUBECONFIG="$1"
 
-# Setup API Authentication
-KUBECTL="oc --config=$KUBECONFIG"
-
-# Setup helper functions
-
 kubectl() {
-	i=0
-
-	echo "Executing kubectl" "$@"
+	(>&2 echo "Executing kubectl $*")
 	while true
 	do
-		i=$((i+1))
-		[ $i -eq 100 ] && echo "kubectl failed, giving up" && exit 1
-
 		set +e
-		out=$($KUBECTL "$@" 2>&1)
+		out=$(oc --config="$KUBECONFIG" "$@" 2>&1)
 		status=$?
 		set -e
 
-		if echo "$out" | grep -q "AlreadyExists"
+		if grep --quiet "AlreadyExists" <<< "$out"
```

Comment:
[POSIX only defines `-q`][1].

And while I'd prefer the here-string form if it was in POSIX, the difference isn't large enough to make it worth breaking with POSIX-compat (for me, anyway).

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/grep.html